<launch>
    <arg name="robot_name" default="lwr"/>
    <arg name="base_frame"  default="link_0"/>
    <arg name="controller_name"  default="orca_ctrl"/>
    <arg name="urdf_url"  default="$(find orca)/examples/resources/lwr.urdf"/>
    <arg name="robot_compensates_gravity" default="true"/>
    <arg name="rviz"  default="true"/>

    <!-- Synchronise the gazebo clock with ROS clock -->
    <param name="/use_sim_time" value="true"/>


    <node pkg="tf2_ros" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0.32 0 0 0 1 link_7 my_super_tool" />

    <node name="gazebo" pkg="orca_ros" type="minimal_controller_gazebo" output="screen">
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="urdf_url" value="$(arg urdf_url)"/>
        <param name="controller_name" value="$(arg controller_name)"/>
        <param name="robot_compensates_gravity" value="$(arg robot_compensates_gravity)"/>
        <rosparam param="init_joint_positions">
            joint_0:  1.0
            joint_3: -1.57
            joint_5:  1.57
        </rosparam>
    </node>

    <rosparam ns="/orca/$(arg robot_name)/$(arg controller_name)" subst_value="true">
      robot_compensates_gravity: $(arg robot_compensates_gravity)
    </rosparam>

    <rosparam ns="/orca/$(arg robot_name)/$(arg controller_name)/CartTask_EE" subst_value="true">
      base_frame: $(arg base_frame)
    </rosparam>

    <node name="rviz_task_controller" pkg="orca_ros" type="rviz_task_controller" output="screen">
        <rosparam subst_value="true">
            robot_name: $(arg robot_name)
            controller_name: $(arg controller_name)
            task_name: CartTask_EE
            interactive_marker_frame_id: $(arg base_frame)
        </rosparam>
    </node>

    <arg name="rviz_config" default="$(find orca_ros)/config/simple_lwr_model.rviz"/>
    <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_config)"/>

    <group ns="/orca">
        <group ns="$(arg robot_name)">

            <param name="robot_description" command="$(find xacro)/xacro '$(arg urdf_url)' --inorder"/>
            <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>
        </group>
    </group>
</launch>
